src/Core/Vipr.Core/CodeModel/OdcmCapabilities.cs

```diff
@@ -0,0 +1,24 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Vipr.Core.CodeModel
+{
+    public class OdcmCapabilities
+    {
+        public bool Insertable { get; set; }
```

Comment:
>"I feel the original model adds an extra layer in converting the raw annotations into OdcmCapability." 

We don't really need an extra level of abstraction here do we? This "Extra level" is going to produce a collection of capability objects which will have roughly the same set of call patterns from the other proposal; it doesn't feel like it makes our lives easier and adds some complexity. 

>"Also OdataCapabiltiesReader right now dispatches the parsing logic to a list of parsers which can parse the annotations any way they want. " 

I don't really agree that this is nescessary. There's not going to be custom parsing semantics for annotations that are already defined in some schema somewhere. The "custom parsing" is just defined as being able to interpret the EDM model, which is parsable fully from knowledge of the types. 

imho the "list of parsers" is just extra code we dont' need + extra complexity.